app/common/state/historyState.js

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const {BrowserWindow} = require('electron')
+const tabState = require('./tabState.js')
+const config = require('../../../js/constants/config.js')
+const eventUtil = require('../../../js/lib/eventUtil.js')
+const appActions = require('../../../js/actions/appActions.js')
+const windowAction = require('../../../js/actions/windowActions')
+const UrlUtil = require('../../../js/lib/urlutil.js')
+const locale = require('../../locale.js')
+const CommonMenu = require('../commonMenu.js')
+
+const onBackButtonHistoryMenu = (tabId, partitionNumber, history, rect) => {
+  const menuTemplate = []
+
+  if (tabId > -1 && history && history.entries.length > 0) {
+    const stopIndex = Math.max(((history.currentIndex - config.navigationBar.maxHistorySites) - 1), -1)
+    for (let index = (history.currentIndex - 1); index > stopIndex; index--) {
+      const url = history.entries[index].url
+
+      menuTemplate.push({
+        label: history.entries[index].display,
+        icon: history.entries[index].icon,
+        click: function (e) {
+          if (eventUtil.isForSecondaryAction(e)) {
+            appActions.createTabRequested({
```

Comment:
this is also going to be a problem because you can't serialize the method call in the windowAction